### PR TITLE
Add endpoint to save list of responses to error report

### PR DESCRIPTION
**Pull request by SemyonKorostik**

### CODE REVIEW

File: src/main/java/com/team4/testingsystem/services/impl/TestEvaluationServiceImpl.java [hunk 1]
```diff
@@ -77,6 +101,10 @@ public void updateScoreAfterCoachCheck(Test test) {
                 coachGrade -> coachGrade.getId().getQuestion().getModule().getName(),
                 coachGrade -> coachGrade));
 
+        List<CoachAnswer> allCoachAnswers = coachAnswerRepository.findAllById_Test(test);
+
+        saveAnswersCoachCheck(Modules.GRAMMAR, allCoachAnswers);
+        saveAnswersCoachCheck(Modules.LISTENING, allCoachAnswers);
```
Comment (tensenrey) [inline src/main/java/com/team4/testingsystem/services/impl/TestEvaluationServiceImpl.java]: modules should be only essay and speaking

File: src/test/java/com/team4/testingsystem/converters/TestVerificationConverterTest.java [hunk 2]
```diff
@@ -1,16 +1,10 @@
...
-import com.team4.testingsystem.dto.QuestionDTO;
-import com.team4.testingsystem.dto.ReportedQuestionDTO;
-import com.team4.testingsystem.dto.TestVerificationDTO;
-import com.team4.testingsystem.entities.ErrorReport;
-import com.team4.testingsystem.entities.Level;
-import com.team4.testingsystem.entities.Question;
-import com.team4.testingsystem.entities.Test;
-import com.team4.testingsystem.entities.TestQuestionID;
+import com.team4.testingsystem.dto.*;
+import com.team4.testingsystem.entities.*;
```
Comment (tensenrey) [inline src/test/java/com/team4/testingsystem/converters/TestVerificationConverterTest.java]: please avoid package imports

File: src/test/java/com/team4/testingsystem/services/impl/ModuleCoachAnswerServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
+package com.team4.testingsystem.services.impl;
+
+import com.team4.testingsystem.entities.*;
```
Comment (tensenrey) [inline src/test/java/com/team4/testingsystem/services/impl/ModuleCoachAnswerServiceImplTest.java]: please avoid package imports

File: src/test/java/com/team4/testingsystem/services/impl/TestEvaluationServiceImplTest.java [hunk 4]
```diff
@@ -1,14 +1,15 @@
 package com.team4.testingsystem.services.impl;
 
 
-import com.team4.testingsystem.entities.Answer;
-import com.team4.testingsystem.entities.ChosenOption;
-import com.team4.testingsystem.entities.CoachGrade;
+import com.team4.testingsystem.entities.*;
```
Comment (tensenrey) [inline src/test/java/com/team4/testingsystem/services/impl/TestEvaluationServiceImplTest.java]: please avoid package imports

File: src/test/java/com/team4/testingsystem/services/impl/ModuleCoachAnswerServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    @org.junit.jupiter.api.Test
+    public void addAnswers()  {
+
+        Mockito.when(moduleService.getModuleByName(module.getName())).thenReturn(module);
+
+        coachAnswer = new CoachAnswer(new TestQuestionID(EntityCreatorUtil.createTest(user, level),
+                EntityCreatorUtil.createQuestion()), COACH_COMMENT);
+
+        Assertions.assertDoesNotThrow(() -> moduleCoachAnswerService.addAnswers(module.getName(), List.of(coachAnswer)));
```
Comment (tensenrey) [inline src/test/java/com/team4/testingsystem/services/impl/ModuleCoachAnswerServiceImplTest.java]: why do you need assertDoesNotThrow?

**Review by tensenrey — CHANGES_REQUESTED** [review]

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]
Please, fix the comments

File: src/main/java/com/team4/testingsystem/services/impl/TestEvaluationServiceImpl.java [hunk 6]
```diff
@@ -80,4 +104,12 @@ public void updateScoreAfterCoachCheck(Test test) {
         saveScoreCoachCheck(test, Modules.ESSAY, gradeMap);
         saveScoreCoachCheck(test, Modules.SPEAKING, gradeMap);
     }
+
+    @Override
+    public void updateCoachAnswersAfterCheck(Test test) {
+        List<CoachAnswer> allCoachAnswers = coachAnswerRepository.findAllById_Test(test);
+
+        saveAnswersCoachCheck(Modules.GRAMMAR, allCoachAnswers);
+        saveAnswersCoachCheck(Modules.LISTENING, allCoachAnswers);
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/services/impl/TestEvaluationServiceImpl.java]: What coach comments do you expect on grammar or listening modules?

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]

File: src/main/java/com/team4/testingsystem/services/impl/TestEvaluationServiceImpl.java (hunk 6 above)
Comment (SemyonKorostik) [reply src/main/java/com/team4/testingsystem/services/impl/TestEvaluationServiceImpl.java]: I'm a little confused) I'll return everything as it was and use the controller.

**Review by AntonMaksimenkau — APPROVED** [review]

**Review by maxim12321 — APPROVED** [review]

**Pull request merged**